src/PieChart/PieArcSeries/PieArcIcon.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import chroma from 'chroma-js';
+
+import type { ArcData } from './PieArcSeries';
+
+interface InjectedProps {
```

Comment:
Interesting! This could be beneficial everywhere!

Reply:
I like it because it helps to identify what the user can pass vs what the system passes that they shouldn't override. 

However, after thinking about it if you want to override low level things and use them indepdently you probably want to be able to pass them all so maybe not?

Reply:
Yes, I have planned that the user may override injected props, but now I understand that this may not be possible :sweat_smile: 

My goal was to make injected props required internally, but optional for the user, that imports component but instead of that I've made that props required for everyone (because of `Props` but optional if you use `PieArcIconProps`.

I'll look if it is possible to achieve such goal. If it is not possible, I'll try to invert the things

Reply:
Ok. I still think that is possible to get correct typings, but it's complicated and I don't know whether the result will be accepted. 

So in this PR I'll leave separation of Props but currently only resulting props will be used (e.g. `PieChartProps`), so the end result will stay the same as before refactoring